src/c++/perf_analyzer/infer_data_manager.cc

```diff
@@ -85,111 +185,19 @@ InferDataManager::InitInferDataOutput(
 
 cb::Error
 InferDataManager::UpdateInputs(
-    const int stream_index, const int step_index, InferData& infer_data)
+    const size_t thread_id, const int stream_index, const int step_index,
+    InferData& infer_data)
 {
   // Reset inputs for this inference request
   infer_data.valid_inputs_.clear();
 
   for (const auto& input : infer_data.inputs_) {
-    RETURN_IF_ERROR(input->Reset());
-
-    const auto& model_input = (*(parser_->Inputs()))[input->Name()];
-
-    const uint8_t* data_ptr{nullptr};
-    size_t batch1_bytesize;
-    const int* set_shape_values = nullptr;
```

Comment:
What happened to this? Why does it seem to only exist for outputs not inputs now? I know this is very specific, but there seems to be a lot refactoring going on. Do we have testing for all of the refactor changes?

Reply:
Yes we have plenty of tests that confirm the raw/shared data in the inputs that get sent to the mock backend.

All of this used to be done every request, which was the cause of all of the overhead. Now it is done for all possible inputs at time 0 in init().